File: SparkExample2/src/main/scala/com/myCompany/scalaExample/SparkHiveConnect.scala
package com.myCompany.scalaExample

import org.apache.spark.{SparkConf, SparkContext}

object SparkHiveConnect extends App {
  case class Person(firstName: String, lastName: String, gender: String)
  val conf = new SparkConf()
    .setAppName("SparkHiveConnect")
    .setMaster("local[1]")

  val sc = new SparkContext(conf)

  val hc = new org.apache.spark.sql.hive.HiveContext(sc)
  import hc.implicits._

  val personResult = sc.textFile("./input/person.txt")
  val personRDD = personResult.map(_.split(",")).map(p => Person(p(0), p(1), p(2)))
  val person = personRDD.toDF
  person.show();
  person.registerTempTable("person1")
  val males = hc.sql("select * from person1")
  males.map(t => "Name: " + t(0)).collect().foreach(println)

  val options = Map("path" -> "hdfs://0.0.0.0:9000/user/hive/warehouse/")
  //person.write.format("orc").options(options).mode("overwrite").saveAsTable("person_orc_New")

  person.write.format("orc").mode("overwrite").saveAsTable("person_orc_New1")
  sc.stop()
}

File: SparkWorkspace/SparkJavaExamples/src/main/java/com/myCompany/javaExample/SparkHiveConnect.java
package com.myCompany.javaExample;

import java.util.Map;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;
import org.apache.spark.sql.DataFrame;
import org.apache.spark.sql.SaveMode;
import org.apache.spark.sql.hive.HiveContext;

import com.myCompany.javaExample.Utils.Person;

public class SparkHiveConnect {

	public static void main(String[] args) {

		SparkConf sparkConf = new SparkConf().setMaster("local").setAppName("SparkHiveConnect");

		JavaSparkContext sc = new JavaSparkContext(sparkConf);

		HiveContext hiveContext = new org.apache.spark.sql.hive.HiveContext(sc);

		JavaRDD<Person> peopleRDD = sc.textFile("./input/person.txt").map(new Function<String, Person>() {
			public Person call(String line) throws Exception {
				String[] parts = line.split(",");

				Person person = new Person();
				person.setFirstName(parts[0]);
				person.setLastName(parts[1]);
				person.setGender(parts[2]);
				return person;
			}
		});

		DataFrame personDataFrame = hiveContext.createDataFrame(peopleRDD, Person.class);
		
		personDataFrame.show();
		
		
		personDataFrame.write().format("orc").mode(SaveMode.Overwrite)
				.option("path", "hdfs://0.0.0.0:9000/user/hive/warehouse/")
				.saveAsTable("PersonJava");
	}
}
